### PR TITLE
[BOLT] Revise IDE folder structure

### PR DESCRIPTION
**Pull request by Meinersbur**
Reviewers of #89153 suggested to break up the patch into per-subproject patches. This is the BOLT part. See #89153 for the entire series and motivation.

Update the folder titles for targets in the monorepository that have not seen taken care of for some time. These are the folders that targets are organized in Visual Studio and XCode (`set_property(TARGET <target> PROPERTY FOLDER "<title>")`) when using the respective CMake's IDE generator.

 * Ensure that every target is in a folder
 * Use a folder hierarchy with each LLVM subproject as a top-level folder
 * Use consistent folder names between subprojects
 * When using target-creating functions from AddLLVM.cmake, automatically deduce the folder. This reduces the number of `set_property`/`set_target_property`, but are still necessary when `add_custom_target`, `add_executable`, `add_library`, etc. are used. A LLVM_SUBPROJECT_TITLE definition is used for that in each subproject's root CMakeLists.txt.

### CODE REVIEW

**Review by rafaelauler — APPROVED** [review]
LGTM

**Review by dcci — APPROVED** [review]

File: bolt/CMakeLists.txt [hunk 1]
```diff
@@ -121,7 +123,7 @@ option(BOLT_BUILD_TOOLS
   "Build the BOLT tools. If OFF, just generate build targets." ON)
 
 add_custom_target(bolt)
-set_target_properties(bolt PROPERTIES FOLDER "BOLT")
+set_target_properties(bolt PROPERTIES FOLDER "BOLT/Meta")
```
Comment (ayermolo) [inline bolt/CMakeLists.txt:126]: This probably should be something more neutral then a company name.

**Comment by aaupov** [comment]
Please use [BOLT] in the title.

File: bolt/CMakeLists.txt (hunk 1 above)
Comment (Meinersbur) [reply bolt/CMakeLists.txt:126]: This is not supposed to be the company name, but meta in the sense of [self-referential](https://en.wikipedia.org/wiki/Meta_(prefix)). That is, targets that only exists to trigger the build of other targets. In this case, all of BOLT (as in `make all` if BOLT was its own build system). "Metatarget" would disambiguate the meaning.

An alternative could be "[Phony](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)".

**Pull request merged**